packages/my-queries-storage/src/hooks.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect } from 'react';
+import { useSyncExternalStore } from 'use-sync-external-store/shim';
+import type { FavoriteQuerySchema, RecentQuerySchema } from './query-storage';
+import { FavoriteQueryStorage, RecentQueryStorage } from './query-storage';
+
+let lastId = 0;
+const subscriptions = new Map<number, () => void>();
+const subscribe = (callback: () => void) => {
+  const subscriptionId = lastId++;
+  subscriptions.set(subscriptionId, callback);
+
+  return () => subscriptions.delete(subscriptionId);
+};
+
+const favoriteStorage = new FavoriteQueryStorage();
+const recentsStorage = new RecentQueryStorage();
+
+const state = {
+  favoriteQueries: [] as Array<typeof FavoriteQuerySchema>,
+  recentQueries: [] as Array<typeof RecentQuerySchema>,
+  loaded: false,
+};
+
+const setFavoriteQueries = (value: Array<typeof FavoriteQuerySchema>) =>
+  (state.favoriteQueries = value as any);
+const setRecentQueries = (value: Array<typeof RecentQuerySchema>) =>
+  (state.recentQueries = value as any);
+
+const notify = () => {
+  for (const cb of subscriptions.values()) {
+    cb();
+  }
+};
+
+export function useSavedQueries() {
+  if (!state.loaded) {
+    state.loaded = true;
+
+    void favoriteStorage
+      .loadAll()
+      .then((q) => setFavoriteQueries(q as any))
+      .then(notify);
+    void recentsStorage
+      .loadAll()
+      .then((q) => setRecentQueries(q as any))
+      .then(notify);
+  }
+
+  const favQueriesSnapshot = useSyncExternalStore(
+    subscribe,
+    () => state.favoriteQueries
+  );
+  const recentQueriesSnapshot = useSyncExternalStore(
+    subscribe,
+    () => state.recentQueries
+  );
```

Comment:
The way subscriptions happen in this hooks needs to be more granular to be usable in Compass, this is kinda okay for this particular store but we do know that this will just not scale for other use-cases like collections or databases lists of a cluster. I think that even for a bare-bones implementation we should put some effort into making it at least somewhat performant because this would affect the public interface too (the idea was to rewrite it later, but keep the interface the same, right?)